android/KMEA/app/src/main/java/com/keyman/engine/KMKeyboardJSHandler.java

```diff
@@ -75,7 +75,10 @@ public String initialKeyboard() {
     Keyboard kbd = Keyboard.getDefaultKeyboard(context);
     int index = prefs.getInt(KMManager.KMKey_UserKeyboardIndex, -1);
     if (index >= 0) {
-      kbd = KMManager.getKeyboardInfo(this.context, index);
+      // It is possible for this method to return `null` - especially
+      // if the KeyboardController instance hasn't yet been initialized.
+      Keyboard lastKbd = KMManager.getKeyboardInfo(this.context, index);
+      kbd = lastKbd != null ? lastKbd : kbd;
```

Comment:
There are no matching error reports that I could find within Sentry, by the way.  That said, this is the only logical way I could determine for an error to be generated within this method to trigger the Web Sentry event; some of the methods involved do have `null` return paths.

Reply:
Does this mean that this function will return information for the wrong keyboard sometimes? Does that just shift the problem sideways rather than resolving it?

Reply:
There's a specific point at which the app and keyboard initialize the `KeyboardController` instance, which is managed as a singleton.  I don't see a hard-and-fast guarantee that the code above won't run before that instance is initialized - and if it isn't, that's when we get a `null`.

I don't think it means we get info for the wrong keyboard - we just don't get info for _any_ keyboard.  The WebView will still send a request for the keyboard later, at a later point after the host page has loaded (and after that initialization has had ample time to complete).  That's my understanding of the situation, anyway.

Reply:
If we get no info for any keyboard, won't we get an exception on line 83 below when we call `kbd.toStub()`?

Reply:
Oh wait, now I see what you mean.  If we don't get a "previous" keyboard, we just get the "default" keyboard, resetting selection to whatever acts as the default.  Later in the load, we can try again for the "last loaded keyboard".